[WIP] sunxi-6.10: Fix broken PWM_SUNXI_ENHANCE driver

chraac opened the pull request:
# Description

Background

In PR #6617, we've added the ethernet and pwm driver for zero2w for kernel version 6.6 and 6.7, now in 6.10, the pwm drvier interface have been changed that makes our driver failed to compile, here's a PR to fix that.

Changes:
- Fix compiling error at the patch `drivers-pwm-Add-pwm-sunxi-enhance-driver-for-h616.patch`, adapt to new 6.10 pwm interface
- At sunxi64 edge config, enable `CONFIG_PWM_SUNXI_ENHANCE` as module

# How Has This Been Tested?
- [x] Build and install to `zero2w`, the ethernet works well
    ![0cd14aa19aef5ad080ac43c408bf7dc](https://github.com/user-attachments/assets/2109ce79-de93-4e71-969c-e772e2d709c6)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

The-going commented:
@chraac It's all good.
Your code is working. Congratulations.

But, but.

1) Your editor is set to indent 4 spaces, not a tab character.
We are correcting the kernel code, which assumes that the indentation is a tab character.
A space is a space and it is not used for indentation at the beginning of a line.

2) It is wrong to add patches to the megous series.
This series is extracted from the repository of the same name and your patches are not there.
They will simply disappear the next time they are extracted for the 6.11 kernel.

The following recommendations make the work easier:

- Do not try to place a large number of changes in different files in one patch,
  even if these changes are related in essence.
  Instead, create your own themed series.
- I'm sure you have a kernel repository: https://git.kernel.org /.
  Extract the linux-6.10.y branch to the working directory.
  Apply as `git am filename.patch` command the existing series of patches in a loop by reading the `series.conf` file.
- Next, make a commit and write small comments to the message in the commit
   why the changes were made in this way.
- If the patch already exists and some of the files require changes and some remain unchanged,
  please feel free to make your own commit and split the existing patch file.
 
When the work is finished:
- Create a new folder `patch/kernel/archive/sunxi-6.10/patches.h616(8)` for example.
- Extract a series of patches to this folder using the [mk_format_patch](https://github.com/armbian/build/blob/main/tools/mk_format_patch) script.
for example:
```
armbian/build> KT=/gituser/linux-stable
armbian/build> PP=patch/kernel/archive/sunxi-6.10
armbian/build> ./tools/mk_format_patch $KT e1a11203a3d9e7220d65..HEAD $PP/patches.h616(8)
```
The script will extract commits from the specified range and create a series file

The-going commented:
I've already done something here https://github.com/armbian/build/pull/7127.
You can just continue.
I can give you access to my repository.

chraac commented:
> @chraac It's all good. Your code is working. Congratulations.
> 
> But, but.
> 
> 1. Your editor is set to indent 4 spaces, not a tab character.
>    We are correcting the kernel code, which assumes that the indentation is a tab character.
>    A space is a space and it is not used for indentation at the beginning of a line.
> 2. It is wrong to add patches to the megous series.
>    This series is extracted from the repository of the same name and your patches are not there.
>    They will simply disappear the next time they are extracted for the 6.11 kernel.
> 
> The following recommendations make the work easier:
> 
> * Do not try to place a large number of changes in different files in one patch,
>   even if these changes are related in essence.
>   Instead, create your own themed series.
> * I'm sure you have a kernel repository: https://git.kernel.org /.
>   Extract the linux-6.10.y branch to the working directory.
>   Apply as `git am filename.patch` command the existing series of patches in a loop by reading the `series.conf` file.
> * Next, make a commit and write small comments to the message in the commit
>   why the changes were made in this way.
> * If the patch already exists and some of the files require changes and some remain unchanged,
>   please feel free to make your own commit and split the existing patch file.
> 
> When the work is finished:
> 
> * Create a new folder `patch/kernel/archive/sunxi-6.10/patches.h616(8)` for example.
> * Extract a series of patches to this folder using the [mk_format_patch](https://github.com/armbian/build/blob/main/tools/mk_format_patch) script.
>   for example:
> 
> ```
> armbian/build> KT=/gituser/linux-stable
> armbian/build> PP=patch/kernel/archive/sunxi-6.10
> armbian/build> ./tools/mk_format_patch $KT e1a11203a3d9e7220d65..HEAD $PP/patches.h616(8)
> ```
> 
> The script will extract commits from the specified range and create a series file

Thahk you for you time to review this PR, 
 

> @chraac It's all good. Your code is working. Congratulations.
> 
> But, but.
> 
> 1. Your editor is set to indent 4 spaces, not a tab character.
>    We are correcting the kernel code, which assumes that the indentation is a tab character.
>    A space is a space and it is not used for indentation at the beginning of a line.
> 2. It is wrong to add patches to the megous series.
>    This series is extracted from the repository of the same name and your patches are not there.
>    They will simply disappear the next time they are extracted for the 6.11 kernel.
> 
> The following recommendations make the work easier:
> 
> * Do not try to place a large number of changes in different files in one patch,
>   even if these changes are related in essence.
>   Instead, create your own themed series.
> * I'm sure you have a kernel repository: https://git.kernel.org /.
>   Extract the linux-6.10.y branch to the working directory.
>   Apply as `git am filename.patch` command the existing series of patches in a loop by reading the `series.conf` file.
> * Next, make a commit and write small comments to the message in the commit
>   why the changes were made in this way.
> * If the patch already exists and some of the files require changes and some remain unchanged,
>   please feel free to make your own commit and split the existing patch file.
> 
> When the work is finished:
> 
> * Create a new folder `patch/kernel/archive/sunxi-6.10/patches.h616(8)` for example.
> * Extract a series of patches to this folder using the [mk_format_patch](https://github.com/armbian/build/blob/main/tools/mk_format_patch) script.
>   for example:
> 
> ```
> armbian/build> KT=/gituser/linux-stable
> armbian/build> PP=patch/kernel/archive/sunxi-6.10
> armbian/build> ./tools/mk_format_patch $KT e1a11203a3d9e7220d65..HEAD $PP/patches.h616(8)
> ```
> 
> The script will extract commits from the specified range and create a series file

chraac commented:
> @chraac It's all good. Your code is working. Congratulations.
> 
> But, but.
> 
> 1. Your editor is set to indent 4 spaces, not a tab character.
>    We are correcting the kernel code, which assumes that the indentation is a tab character.
>    A space is a space and it is not used for indentation at the beginning of a line.
> 2. It is wrong to add patches to the megous series.
>    This series is extracted from the repository of the same name and your patches are not there.
>    They will simply disappear the next time they are extracted for the 6.11 kernel.
> 
> The following recommendations make the work easier:
> 
> * Do not try to place a large number of changes in different files in one patch,
>   even if these changes are related in essence.
>   Instead, create your own themed series.
> * I'm sure you have a kernel repository: https://git.kernel.org /.
>   Extract the linux-6.10.y branch to the working directory.
>   Apply as `git am filename.patch` command the existing series of patches in a loop by reading the `series.conf` file.
> * Next, make a commit and write small comments to the message in the commit
>   why the changes were made in this way.
> * If the patch already exists and some of the files require changes and some remain unchanged,
>   please feel free to make your own commit and split the existing patch file.
> 
> When the work is finished:
> 
> * Create a new folder `patch/kernel/archive/sunxi-6.10/patches.h616(8)` for example.
> * Extract a series of patches to this folder using the [mk_format_patch](https://github.com/armbian/build/blob/main/tools/mk_format_patch) script.
>   for example:
> 
> ```
> armbian/build> KT=/gituser/linux-stable
> armbian/build> PP=patch/kernel/archive/sunxi-6.10
> armbian/build> ./tools/mk_format_patch $KT e1a11203a3d9e7220d65..HEAD $PP/patches.h616(8)
> ```
> 
> The script will extract commits from the specified range and create a series file

Thank you for your review. appreciate your feedback!
1. Regarding point 1: The original patch comes from the Sunxi open-source repository, which uses space indentation. To minimize changes and maintain consistency with the source, I kept the space indentation. Also, since this is a single-function device driver, I merged it into a single patch for simplicity.
2. Regarding point 2: I'm happy to add the patch to your PR if that works better for you.

The-going commented:
> * Regarding point 1: The original patch comes from the Sunxi open-source repository, which uses space indentation. To minimize changes and maintain consistency with the source, I kept the space indentation. Also, since this is a single-function device driver, I merged it into a single patch for simplicity.

In fact, PR people react negatively to such remarks. They wait in silence.
You answer politely. Thank you for that.

> * Regarding point 2: I'm happy to add the patch to your PR if that works better for you.

The build system applies patches using its own method. He is less strict.
It will try to apply in any case, even with a large offset or diffusion of strings.
Sometimes this leads to strange errors when the changes turn out to be in
the wrong function or in the wrong node for the DTS.

The only way that works flawlessly for me is to apply the entire series to the latest version
of the kernel repository, and then re-extract the entire series.
This ensures that patches are applied cleanly in the build system itself.

In this sense, it would be better if your fixes were directly included in the armbian patch series for  sunxi-6.10.
Only for version 6.10.
You can simply mark existing patches with a minus sign to disable and add new ones to the end of the list.
I will pull out your changes and re-export the patches.
Those marked as unused will be automatically deleted.

P.S.
In fact, I have brought very few real changes.
I'm just moving patches to a new version and I can't always check the quality of the patch itself.
You can see a lot of patches in which I am listed as a commiter and there are no comments on the commit message.
This means only one thing, someone once added a patch extracted as `git diff` or `diff -u` .....

With respect.

chraac commented:
> > * Regarding point 1: The original patch comes from the Sunxi open-source repository, which uses space indentation. To minimize changes and maintain consistency with the source, I kept the space indentation. Also, since this is a single-function device driver, I merged it into a single patch for simplicity.
> 
> In fact, PR people react negatively to such remarks. They wait in silence. You answer politely. Thank you for that.
> 
> > * Regarding point 2: I'm happy to add the patch to your PR if that works better for you.
> 
> The build system applies patches using its own method. He is less strict. It will try to apply in any case, even with a large offset or diffusion of strings. Sometimes this leads to strange errors when the changes turn out to be in the wrong function or in the wrong node for the DTS.
> 
> The only way that works flawlessly for me is to apply the entire series to the latest version of the kernel repository, and then re-extract the entire series. This ensures that patches are applied cleanly in the build system itself.
> 
> In this sense, it would be better if your fixes were directly included in the armbian patch series for sunxi-6.10. Only for version 6.10. You can simply mark existing patches with a minus sign to disable and add new ones to the end of the list. I will pull out your changes and re-export the patches. Those marked as unused will be automatically deleted.
> 
> P.S. In fact, I have brought very few real changes. I'm just moving patches to a new version and I can't always check the quality of the patch itself. You can see a lot of patches in which I am listed as a commiter and there are no comments on the commit message. This means only one thing, someone once added a patch extracted as `git diff` or `diff -u` .....
> 
> With respect.

yeah, gonna close this PR and try to merge the pwm driver into your branch
